File: socialgo/src/main/java/com/fungo/socialgo/listener/OnShareListener.java
package com.fungo.socialgo.listener;

import com.fungo.socialgo.exception.SocialError;
import com.fungo.socialgo.model.ShareObj;

/**
 * CreateAt : 2016/12/25
 * Describe : 分享监听
 *
 * @author chendong
 */

public interface OnShareListener {

    void onStart(int shareTarget, ShareObj obj);

    /**
     * 准备工作，在子线程执行
     *
     * @param shareTarget 分享目标
     * @param obj         shareMediaObj
     */
    ShareObj onPrepareInBackground(int shareTarget, ShareObj obj) throws Exception;

    void onSuccess();

    void onFailure(SocialError e);

    void onCancel();
}

File: socialgo/src/main/java/com/fungo/socialgo/listener/OnShareListener.kt
package com.fungo.socialgo.listener

import com.fungo.socialgo.exception.SocialError
import com.fungo.socialgo.model.ShareEntity

/**
 * 分享的回调
 */
interface OnShareListener {

    fun onStart(shareTarget: Int, obj: ShareEntity)

    /**
     * 准备工作，在子线程执行
     */
    @Throws(Exception::class)
    fun onPrepareInBackground(shareTarget: Int, obj: ShareEntity): ShareEntity?

    fun onSuccess()

    fun onFailure(e: SocialError)

    fun onCancel()
}
